lib/extractor/layer.ts

```diff
@@ -53,6 +55,70 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: OciArchiveManifest[] = [];
+    let imageIndex;
+
+    tarExtractor.on("entry", async (header, stream, next) => {
+      if (header.type === "file") {
+        if (isImageIndexFile(header.name)) {
+          imageIndex = await streamToJson<OciImageIndex>(stream);
+        } else {
+          const streamCopy1 = new PassThrough();
+          const streamCopy2 = new PassThrough();
+          stream.pipe(streamCopy1);
+          stream.pipe(streamCopy2);
+
+          const promises = [
+            streamToJson(streamCopy1).catch(() => undefined),
+            extractImageLayer(streamCopy2, extractActions).catch(
+              () => undefined,
+            ),
+          ];
+          const [manifest, layer] = await Promise.all(promises);
+
+          const headerParts = header.name.split("/");
+          const headerName = `sha256:${headerParts[headerParts.length - 1]}`;
```

Comment:
`header.name` in OCI archives is something like: `blob/sha256/xxxxxxxx`, so I'm extracting here `sha256:xxxxxxx`, which makes it super easy then to compare it to the layer identifier in manifests, which is of this exact format, wdyt??

Reply:
is it always sha256? or can be other hashes?

Reply:
I think the two `consts` can be in the `if (layer !== undefined) {` flow

Reply:
I've tried a few images and it was always `sha256`, BUT looking now its'a a digest and according to the spec: https://github.com/opencontainers/image-spec/blob/master/descriptor.md#digests it can be different hashes too. Hmmm, ok I think I can extract which hash it is from the path, which is `/blob/sha256/xxxxxx` in this case 🤔

Reply:
Based on above documentation I've extracted format: `hash_name:hash_value`, let me know what you think!